caffe2/serialize/inline_container.cc

```diff
@@ -42,27 +46,31 @@ static std::string basename(const std::string& name) {
 }
 
 size_t PyTorchStreamReader::read(uint64_t pos, char* buf, size_t n) {
-  in_->seekg(pos);
-  if(!*in_)
-    return 0;
-  in_->read(static_cast<char*>(buf), n);
-  if(!*in_)
-    return 0;
-  return n;
+  return in_->read(pos, buf, n, "reading file");
+}
+
+PyTorchStreamReader::PyTorchStreamReader(std::string file_name, std::istream* in) {
+  if (!in) {
+    in_ = std::make_shared<at::FileAdapter>(file_name);
+  } else {
+    in_ = std::make_shared<at::IStreamAdapter>(in);
+  }
+  init();
 }
 
-PyTorchStreamReader::PyTorchStreamReader(std::string file_name, std::istream* in)
-: ar_(new mz_zip_archive), in_(in) {
+PyTorchStreamReader::PyTorchStreamReader(std::shared_ptr<at::ReadAdapterInterface> in) {
+  in_ = in;
+  init();
+}
+
+void PyTorchStreamReader::init() {
+  AT_ASSERT(in_ != nullptr);
+  ar_ = std::unique_ptr<mz_zip_archive>(new mz_zip_archive);
   memset(ar_.get(), 0, sizeof(mz_zip_archive));
 
-  if (!in_) {
-    file_stream_.open(file_name, std::ifstream::in | std::ifstream::binary);
-    in_ = &file_stream_;
-    valid("opening archive");
-  }
+  valid("opening archive");
```

Comment:
this one doesn't help any more (as `valid` only operates on miniz operations). remove?